Job check actions fails with "fatal: Need to specify how to reconcile divergent branches."

Motivation
----------
Multiple jobs failed with the error message (e.g. https://github.com/rucio/rucio/runs/4948366129?check_suite_focus=true)
```
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
From https://github.com/rucio/rucio
 * branch                master     -> FETCH_HEAD
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint: 
hint:   git config pull.rebase false  # merge (the default strategy)
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.
Error: Process completed with exit code 128.
```

This is due to the fact that the `git pull origin master` not only pulls the master branch from origin, but also tries to merge the pulled master branch with the current one (https://www.git-scm.com/docs/git-pull#_examples). This leads to the divergent branches' error.

Modification
------------
The `git pull origin master` is not needed anymore, since the `fetch-depth: 0` pulls all branches and tags (https://github.com/actions/checkout#usage). Just delete the line.

--- a/.github/workflows/autotest.yml
+++ b/.github/workflows/autotest.yml
@@ -15,7 +15,6 @@ jobs:
       - name: Check headers
         shell: bash
         run: |
-          git pull origin master
           echo "Current git rev $(git rev-parse HEAD)"
           echo "Common ancestor git rev $(git merge-base HEAD remotes/origin/master)"
           echo "If a file header needs to be changed run 'tools/add_header -i FILE' from the rucio project directory."

